packages/venia-concept/src/components/MiniCart/product.js

```diff
@@ -1,161 +1,116 @@
-                    label: string,
-                    value: string
-                })
-            ),
-            price: number.isRequired,
-            product_type: string,
-            qty: number.isRequired,
-            quote_id: string,
-            sku: string.isRequired
-        }).isRequired,
-        currencyCode: string.isRequired,
-        openOptionsDrawer: func.isRequired
-    };
+const Product = props => {
+    const { beginEditItem, currencyCode, item, removeItemFromCart } = props;
+    const { image, name, options, price, qty } = item;
 
-    // TODO: Manage favorite items using GraphQL/REST when it is ready
-    constructor() {
-        super();
-        this.state = {
-            isLoading: false,
-            isFavorite: false
-        };
-    }
+    const imageRef = useRef(null);
+    const [backgroundImage, setBackgroundImage] = useState(null);
+    const [isFavorite, setIsFavorite] = useState(false);
+    const [isLoading, setIsLoading] = useState(false);
 
-    get options() {
-        const { classes, item } = this.props;
-        const options = item.options;
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const mask = isLoading ? <div className={classes.mask} /> : null;
 
-        return options && options.length > 0 ? (
-            <dl className={classes.options}>
-                {options.map(({ label, value }) => (
-                    <Fragment key={`${label}${value}`}>
-                        <dt className={classes.optionLabel}>
-                            {label} : {value}
-                        </dt>
-                    </Fragment>
-                ))}
-            </dl>
-        ) : null;
-    }
+    useEffect(() => {
+        // After the imageRef mounts, get its --image-width so
+        // we can pass it to `resourceUrl`.
+        const imageWidth = getComputedStyle(imageRef.current).getPropertyValue(
+            '--image-width'
+        );
 
-    get mask() {
-        const { classes } = this.props;
-        return this.state.isLoading ? <div className={classes.mask} /> : null;
-    }
+        if (imageWidth) {
+            // parseFloat to strip off any units like 'px'.
+            const parsedImageWidth = parseFloat(imageWidth);
+            setBackgroundImage(
+                `url(${resourceUrl(image.file, {
+                    type: 'image-product',
+                    width: parsedImageWidth
+                })})`
+            );
+        }
+    }, [imageRef]);
 
-    styleImage(image) {
-        return {
-            minHeight: imageHeight, // min-height instead of height so image will always align with grid bottom
-            width: imageWidth,
-            backgroundImage: `url(${resourceUrl(image.file, {
-                type: 'image-product',
-                width: imageWidth
-            })})`
-        };
-    }
+    const handleFavoriteItem = useCallback(() => {
+        setIsFavorite(!isFavorite);
+    }, [isFavorite, setIsFavorite]);
```

Comment:
Yeah, I'll admit I'm a little fuzzy on why some things can be omitted from this list.

Reply:
> It is only safe to omit a function from the dependency list if nothing in it (or the functions called by it) references props, state, or values derived from them. 

Read more [here](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies). Since state hooks do not reference prop, state or values derived by the hook itself (as in, doesn't change on re-render) you can omit.

Also they even note it for you [in the docs](https://reactjs.org/docs/hooks-reference.html) but it's not easy to find: 

> React guarantees that setState function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.

Reply:
Resolved by https://github.com/magento-research/pwa-studio/pull/1266/commits/fb51f2cef83a2e83e26ba4a5f3cc35ba757d8ae9.